hazelcast/proxy/multi_map.py

```diff
@@ -445,6 +447,52 @@ def put(self, key: KeyType, value: ValueType) -> Future[bool]:
         request = multi_map_put_codec.encode_request(self.name, key_data, value_data, thread_id())
         return self._invoke_on_key(request, key_data, multi_map_put_codec.decode_response)
 
+    def put_all(self, map: typing.Dict[KeyType, typing.Sequence[ValueType]]) -> Future[bool]:
+        """Stores given values by associating with corresponding keys in dictionary format.
+
+        Warning:
+            This method uses ``__hash__`` and ``__eq__`` methods of binary form
+            of the key, not the actual implementations of ``__hash__`` and
+            ``__eq__`` defined in key's class.
+
+        Args:
+            map: the map corresponds to multimap entries.
+
+        Returns:
+            ``True`` if size of the multimap is increased, ``False`` if the
+            multimap already contains the key-value tuple.
+        """
+        check_not_none(map, "map can't be None")
+        if not map:
+            return ImmediateFuture(None)
+
+        partition_service = self._context.partition_service
+        partition_map: typing.Dict[int, typing.List[typing.Tuple[Data, typing.List[Data]]]] = {}
+
+        for key, values in map.items():
+            try:
+                check_not_none(key, "key can't be None")
+                check_not_none(values, "values can't be None")
+                serialized_key = self._to_data(key)
+                serialized_values = []
+                for value in values:
+                    check_not_none(value, "value can't be None")
+                    serialized_values.append(self._to_data(value))
+                partition_id = partition_service.get_partition_id(serialized_key)
+                try:
+                    partition_map[partition_id].append((serialized_key, serialized_values))
+                except KeyError:
+                    partition_map[partition_id] = [(self._to_data(key), serialized_values)]
```

Comment:
changing the partition_map to `defaultdict` add only doing `append` would be more idiomatic

Reply:
Excepting `KeyError` is required I guess. It is not catched by the upper surrounding exception handling.

Reply:
I meant you can make use of this to get rid of KeyeError handling. https://docs.python.org/3/library/collections.html#collections.defaultdict